JINJA tests

alexbleotu opened the pull request:
### What does this PR do?

Added mechanism to register additional jinja tests; added `match`, `equal` jinja tests that can be used standalone, or with the selectattr filter.


### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Commit:
Added decorator and registered additional JINJA tests

Commit:
Added match, equal jinja tests

- can be used with the selectattr filter

salt-jenkins commented:
@alexbleotu, thanks for your PR! By analyzing the history of the files in this pull request, we identified @lorengordon, @s0undt3ch and @rallytime to be potential reviewers.

lorengordon commented:
Might be nice to add a "Tests" section to the jinja topic doc, similar to the "Filters" section.

https://github.com/saltstack/salt/blob/develop/doc/topics/jinja/index.rst#filters

alexbleotu commented:
Sounds good. Wasn't aware of the doc.

Commit:
Renamed equal test to equalto and fixed the comments

Commit:
Added Tests section in JINJA documentation

alexbleotu commented:
Renamed `equal` to `equalto` as I found it in a different tool and though we should be consistent. Also added the Tests section to the documentation. Thanks for pointing it out!

lorengordon commented:
Nice, looks good to me! :D

cachedout reviewed (approved):
I like this. Well done.
